app/models/dialog_field_drop_down_list.rb

```diff
@@ -30,6 +38,22 @@ def refresh_json_value(checked_value)
     {:refreshed_values => refreshed_values, :checked_value => @value, :read_only => read_only?, :visible => visible?}
   end
 
+  def automate_output_value
+    return super unless multi_value?
+    a = @value.chomp.split(',')
+    automate_values = a.first.kind_of?(Integer) ? a.map(&:to_i) : a
+    MiqAeEngine.create_automation_attribute_array_value(automate_values)
+  end
+
+  def automate_key_name
+    MiqAeEngine.create_automation_attribute_array_key(super)
```

Comment:
K, does it also need the  def value_from_dialog_fields(dialog_values) that's in tagControl?